src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -717,7 +717,7 @@ public BoundExpression ComplexConditionalReceiver(BoundExpression valueTypeRecei
 
         public BoundExpression Coalesce(BoundExpression left, BoundExpression right)
         {
-            Debug.Assert(left.Type.Equals(right.Type, TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds | TypeCompareKind.IgnoreNullableModifiersForReferenceTypes));
+            Debug.Assert(left.Type.Equals(right.Type, TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds | TypeCompareKind.IgnoreNullableModifiersForReferenceTypes) || left.Type.IsErrorType());
```

Comment:
>eft.Type.IsErrorType() [](start = 183, length = 22)

The need for this confuses me.  If a program has errors, we don't lower it or perform code generation.  So why would this be needed?

Reply:
This assertion was triggered by the unit test which tests the case where `string.Concat(object)` is missing. I'm not at all convinced that this is the right approach: please let me know if I should do something else.

Reply:
I understand: the error bound node is triggered by an error arising during lowering.  So this is correct.